packages/twenty-server/src/modules/messaging/message-import-manager/drivers/microsoft/services/microsoft-fetch-by-batch.service.ts

```diff
@@ -42,16 +42,45 @@ export class MicrosoftFetchByBatchService {
         },
       }));
 
-      const batchResponse = await client
-        .api('/$batch')
-        .post({ requests: batchRequests });
+      try {
+        const batchResponse = await client
+          .api('/$batch')
+          .post({ requests: batchRequests });
 
-      batchResponses.push(batchResponse);
+        batchResponses.push(batchResponse);
+      } catch (error) {
+        if (this.isTemporaryError(error)) {
+          throw {
+            statusCode: 429,
+            message: error.body,
+            code: error.code,
+          };
+        } else {
+          throw error;
+        }
+      }
     }
 
     return {
       messageIdsByBatch,
       batchResponses,
     };
   }
+
+  /**
+   * Microsoft client.api.post sometimes throws (hard to catch) temporary errors like this one:
```

Comment:
replace this comment by a test (we should extract the util isMicrosoftClientTemporaryError)